python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -0,0 +1,342 @@
+                    self.logger.error("Error loading index file: %s" % str(e))
+                    self.index = None
+                    self.index_builder = file_index.FileIndexBuilder() if generate_index else None
+            else:
+                self.logger.debug("No index file found @ '%s'." % self.index_path)
+                self.index = None
+                self.index_builder = file_index.FileIndexBuilder() if generate_index else None
+
+        if self.index_builder is not None:
+            self.logger.debug("Generating index file '%s'." % self.index_path)
+
+    def have_index(self):
+        return self.index is not None
+
+    def get_index(self):
+        return self.index
+
+    def generating_index(self):
+        return self.index_builder is not None
+
+    def get_bytes_read(self):
+        return self.total_bytes_read
+
+    def next(self):
+        while True:
+            if not self._advance_to_next_sync():
+                # End of file.
+                self.total_bytes_read = self.input_file.tell()
+                self.logger.debug('EOF reached.')
+                break
+
+            offset_bytes = self.input_file.tell()
+            self.total_bytes_read = offset_bytes
+            self._print_progress()
+
+            if self.total_bytes_read - self.last_print_bytes > 10e6:
+                elapsed_sec = (datetime.now() - self.start_time).total_seconds()
+                self.logger.log(logging.INFO if self.show_progress else logging.DEBUG,
+                                'Processed %d/%d bytes (%.1f%%). [elapsed=%.1f sec, rate=%.1f MB/s]' %
+                                (self.total_bytes_read, self.file_size_bytes,
+                                 100.0 * float(self.total_bytes_read) / self.file_size_bytes,
+                                 elapsed_sec, self.total_bytes_read / elapsed_sec / 1e6))
+                self.last_print_bytes = self.total_bytes_read
+
+            self.logger.trace('Reading candidate message @ %d (0x%x).' % (offset_bytes, offset_bytes), depth=2)
+
+            # Read the next message header.
+            data = self.input_file.read(MessageHeader.calcsize())
+            read_len = len(data)
+            if read_len < MessageHeader.calcsize():
+                # End of file.
+                self.logger.debug('EOF reached.')
+                break
+
+            try:
+                header = MessageHeader()
+                header.unpack(data, warn_on_unrecognized=False)
+
+                # Check if the payload is too big.
+                if header.payload_size_bytes > MessageHeader._MAX_EXPECTED_SIZE_BYTES:
+                    raise ValueError('Payload size (%d) too large.' % header.payload_size_bytes)
+
+                # Read and validate the payload.
+                payload_bytes = self.input_file.read(header.payload_size_bytes)
+                read_len += len(payload_bytes)
+                if len(payload_bytes) != header.payload_size_bytes:
```

Comment:
If I understand the two possibilities here are:
1. You reached the end of the file
2. You are near the end of the file, but you got an invalid sync so you are reading an invalid message but don't have enough bytes to validate the CRC so you might miss other valid messages before the end?

If that is the case you're catching it's probably worth having a comment to explain.

Reply:
Yeah good idea, will comment